src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java

```diff
@@ -317,33 +294,48 @@ private int instances(AppDeploymentRequest request) {
 	}
 
 	private Mono<Void> pushApplication(String deploymentId, AppDeploymentRequest request) {
-		return requestPushApplication(PushApplicationRequest.builder()
-			.application(getApplication(request)) // Only one of the two is non-null
+		ApplicationManifest.Builder manifest = ApplicationManifest.builder()
+			.path(getApplication(request)) // Only one of the two is non-null
 			.dockerImage(getDockerImage(request)) // Only one of the two is non-null
 			.buildpack(buildpack(request))
-			.diskQuota(diskQuota(request))
-			.domain(domain(request))
+			.disk(diskQuota(request))
+			.environmentVariables(getEnvironmentVariables(deploymentId, request))
 			.healthCheckType(healthCheck(request))
-			.host(host(request))
 			.instances(instances(request))
 			.memory(memory(request))
 			.name(deploymentId)
 			.noRoute(toggleNoRoute(request))
-			.noStart(true)
-			.routePath(routePath(request))
-			.build())
+			.services(servicesToBind(request));
+
+		String domain = domain(request);
+		String host = host(request);
+		String routePath = routePath(request);
+
+		if (routePath != null) {
+			manifest
+				.route(Route.builder()
+					.route(String.format("%s.%s%s", host, domain, routePath))
```

Comment:
You've ensure that they won't be by [providing](https://github.com/nebhale/spring-cloud-deployer-cloudfoundry/blob/df3c1aa4399896965530a281a1ac0c23f8d37aa9/src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java#L216) [defaults](https://github.com/nebhale/spring-cloud-deployer-cloudfoundry/blob/df3c1aa4399896965530a281a1ac0c23f8d37aa9/src/main/java/org/springframework/cloud/deployer/spi/cloudfoundry/CloudFoundryAppDeployer.java#L287).